### DISCUSSION TITLE
Mobx and Redux in the same app?

### DISCUSSION THREAD

**Discussion by artemis-prime** [mobxjs/mobx#3061] [General] [answered]
If one's goal is to slowly transition a Redux app to Mobx, is there any reason why one could not do it gradually and piecemeal? For example, have both in the system at the same time, go page by page, or store domain by store domain? I've always imagined it should be fine, but have never tried. Thanks for all the outstanding work Michel! Mobx rocks.

**Comment by artemis-prime** [comment]
Michel replied: yeah I'd do it reducer per reducer, and store the observable object that
holds the mobx state at that place in the redux tree instead, so that in
the state tree all the leaves become mobx observable objects piece by piece

(pasted from other question) @mweststrate

**Reply by artemis-prime** [reply]
Why?  Why not just organize the mobx stuff the way I normally would?

**Reply by artemis-prime** [reply]
"so that in the state tree all the leaves become mobx observable objects piece by piece?  "

@mweststrate, what does that achieve?  (My redux knowledge isn't that deep since I've been using that upstart MobX thing for years ;) )

**Comment by mweststrate** [comment, marked as answer]
keeps your redux reducers working for the logic they're responsible for.
But listen, just play with it, experiment some. There are probably a dozen
ways you could approach that, but you have the context on it, anyone here
knows nothing about your concrete code, so any recommendation is going to
be super abstract, and you'll just have to dive into and see where it gets
you to get more informed than what is suggested here.
